packages/connect/src/promise-client.spec.ts

```diff
@@ -45,68 +47,90 @@ const TestService = {
 } as const;
 
 describe("createClientStreamingFn()", function () {
-  const inputLog: string[] = [];
+  it("works as expected on the happy path", async () => {
+    const input = new Int32Value({ value: 1 });
 
-  // eslint-disable-next-line @typescript-eslint/require-await
-  async function* input() {
-    try {
-      inputLog.push("yield 1");
-      yield new Int32Value({
-        value: 1,
+    const output = new StringValue({ value: "yield 1" });
+    const transport = createRouterTransport(({ service }) => {
+      service(TestService, {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars -- needed for inference
+        clientStream: (_?: AsyncIterable<Int32Value>) =>
+          Promise.resolve(output),
       });
-    } finally {
-      inputLog.push("finally");
-    }
-  }
-
-  it("works as expected on the happy path", async () => {
-    const transport = stubTransport({});
+    });
     const fn = createClientStreamingFn(
       transport,
       TestService,
       TestService.methods.clientStream
     );
-    const res = await fn(input());
+    const res = await fn(
+      // eslint-disable-next-line @typescript-eslint/require-await
+      (async function* () {
+        yield input;
+      })()
+    );
     expect(res).toBeInstanceOf(StringValue);
-    expect(inputLog).toEqual(["yield 1", "finally"]);
+    expect(res.value).toEqual(output.value);
   });
 });
 
 describe("createServerStreamingFn()", function () {
   it("works as expected on the happy path", async () => {
-    const transport = stubTransport({
-      streamOutput: [null, null, null],
+    const output = [
+      new StringValue({ value: "input1" }),
+      new StringValue({ value: "input2" }),
+      new StringValue({ value: "input3" }),
+    ];
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- arguments not used for mock
+    const serverStreamImpl = (_input: Int32Value, _context: HandlerContext) =>
+      createAsyncIterable(output);
+    const serverStream = jasmine
+      .createSpy("serverStreamSpy", serverStreamImpl)
```

Comment:
Glad you asked.  I'm really pulling my hair out with the Jasmine matchers.  If jasmine was a [popular](https://npmtrends.com/ava-vs-jasmine-vs-jest-vs-karma-vs-mocha-vs-tape) choice I'd probably push harder, but I think the fact that the matcher output is this incomprehensible kinda speaks for itself, so I focused what I could to work around it.  Here's a good example:

```
1) createClientStreamingFn() works as expected on the happy path
  Message:
    Expected spy clientStream to have been called only once, and with given args:
      [ Object({ next: Function, throw: Function, return: Function, Symbol(Symbol.asyncIterator): Function }), <jasmine.objectContaining(Object({ method: Object({ name: 'ClientStream', I: Function, O: Function, kind: 2 }), service: Object({ typeName: 'handwritten.TestService', methods: Object({ clientStream: Object({ name: 'ClientStream', I: Function, O: Function, kind: 2 }), serverStream: Object({ name: 'ServerStream', I: Function, O: Function, kind: 1 }), bidiStream: Object({ name: 'BidiStream', I: Function, O: Function, kind: 3 }) }) }) }))> ]
    But the actual call was:
      [ Object({ next: Function, throw: Function, return: Function, Symbol(Symbol.asyncIterator): Function }), Object({ method: Object({ name: 'ClientStream', I: Function, O: Function, kind: 2 }), service: Object({ typeName: 'handwritten.TestService', methods: Object({ clientStream: Object({ name: 'ClientStream', I: Function, O: Function, kind: 2 }), serverStream: Object({ name: 'ServerStream', I: Function, O: Function, kind: 1 }), bidiStream: Object({ name: 'BidiStream', I: Function, O: Function, kind: 3 }) }) }), requestHeader: [object Headers], responseHeader: [object Headers], responseTrailer: [object Headers] }) ].
    Expected $[0].next = Function to equal Function.
    Expected $[0].throw = Function to equal Function.
    Expected $[0].return = Function to equal Function.
    Expected $[0][Symbol(Symbol.asyncIterator)] = Function to equal Function.

```

It failed, apparently, because it expected `Function` to equal `Function`.  Thanks, jasmine... heh.  _very helpful_.

I pushed an example of how this looks to 7e1308d151f429501db622c4d40b53e477d3ed1b

Reply:
We need Jasmine for running tests in a browser, including very old ones through browserstack. IIRC, matchers are very basic in Jasmine, but you can provide your own custom equality tester for `Headers`.

But how would you expect a matcher to work on an asynchronous iterable, in general? 

Let's find a good solution for our specific use case with these tests, happy to chat offline.

Reply:
[update] we chatted offline and I think this is resolved now